File: plugins/devkit/devkit-java-tests/testData/inspections/serializableCtor/CorrectAnnotatedConstructor.java
import java.io.Serializable;
import com.intellij.serialization.PropertyMapping;

public class CorrectAnnotatedConstructor implements Serializable {

  private static final long serialVersionUID = 1L;

  final String myString;
  final Integer myInteger;
  final Boolean myBoolean;

  @PropertyMapping({"myString", "myInteger", "myBoolean"})
  public CorrectAnnotatedConstructor(String string, Integer integer, Boolean bool) {
    myString = string;
    myInteger = integer;
    myBoolean = bool;
  }
}

File: plugins/devkit/devkit-kotlin-tests/testData/inspections/serializableCtor/CorrectAnnotatedConstructor.kt
import com.intellij.serialization.PropertyMapping
import java.io.Serializable

class CorrectAnnotatedConstructor @PropertyMapping("myString", "myInteger", "myBoolean") constructor(val myString: String,
                                                                                                     val myInteger: Int,
                                                                                                     val myBoolean: Boolean) : Serializable {
  companion object {
    private const val serialVersionUID = 1L
  }
}